src/csrc/include/internal/rl/on_policy/ppo.h

```diff
@@ -84,25 +84,31 @@ void train_ppo(const ACPolicyPack& pq_model, torch::Tensor state_tensor, torch::
     torch::NoGradGuard no_grad;
 
     // compute mean
-    torch::Tensor adv_mean = torch::mean(adv_tensor);
+    torch::Tensor adv_mean = torch::sum(adv_tensor);
+    auto options = torch::TensorOptions().dtype(torch::kLong).device(adv_mean.device());
+    torch::Tensor adv_count = torch::tensor({torch::numel(adv_tensor)}, options);
 
     // average mean across all nodes
     if (pq_model.comm) {
       std::vector<torch::Tensor> means = {adv_mean};
-      pq_model.comm->allreduce(means, true);
+      pq_model.comm->allreduce(means, false);
```

Comment:
Might be cleaner to allreduce multiple tensors at once like:
```
std::vector<torch::Tensor> tensors = {adv_mean, adv_count};
pq_model.comm->allreduce(tensors, false);
adv_mean = tensors[0];
adv_count = tensors[1];
```
It will wrap the two allreduces in a NCCL group which _could_ be more efficient. Might not matter since the tensors are of different data type but it cleans up the code a little anyway.

Reply:
Ha, I was wondering about that. Does that work?

Reply:
I sure hope so since that is how we allreduce grads. :)
https://github.com/NVIDIA/TorchFort/blob/4670c06d2399841fe93dee5c90a737e6119e3296/src/csrc/training.cpp#L147-L152